pkg/scheduler/algorithm/priorities/metadata.go

```diff
@@ -123,9 +123,25 @@ func getSelector(pod *v1.Pod, sl corelisters.ServiceLister, cl corelisters.Repli
 		}
 	}
 
-	if rcs, err := cl.GetPodControllers(pod); err == nil {
-		for _, rc := range rcs {
-			labelSet = labels.Merge(labelSet, rc.Spec.Selector)
+	rcPod, rsPod, stsPod := false, false, false
+	for _, ref := range pod.OwnerReferences {
+		switch ref.Kind {
+		case "ReplicationController":
+			rcPod = true
+		case "ReplicaSet":
+			rsPod = true
+		case "StatefulSet":
+			stsPod = true
+		default:
+			continue
+		}
+	}
+
+	if rcPod == true {
```

Comment:
rcPod == true
->
rcPod    seems simple

Reply:
yes, thank you